src/gleam/uri.gleam

```diff
@@ -164,3 +168,133 @@ pub fn to_string(uri: Uri) -> String {
   |> dynamic.string
   |> result.unwrap("")
 }
+
+/// Fetch the origin of a uri
+///
+/// Return the origin of a uri as defined in
+/// https://tools.ietf.org/html/rfc6454
+///
+/// The supported uri schemes are `http` and `https`
+/// Urls without a scheme will return Error
+pub fn origin(uri: Uri) -> Result(String, Nil) {
+  let Uri(scheme: scheme, host: host, port: port, ..) = uri
+  case scheme {
+    Some("https") | Some("http") -> {
+      let origin = Uri(scheme, None, host, port, "", None, None)
+      Ok(to_string(origin))
+    }
+    _ -> Error(Nil)
+  }
+}
+
+fn option_or(first: Option(a), second: Option(a)) -> Option(a) {
+  case first {
+    Some(_) -> first
+    None -> second
+  }
+}
+
+fn drop_last(elements) {
```

Comment:
This wouldn't need to be a utility function if list.split took negative arguments, and worked from the reverse in such a case. 

Though I don't mind it being here. that's just a comment.